--- ./tmp/Closure57buggy/src/com/google/javascript/jscomp/ClosureCodingConvention.java	2019-05-05 07:50:53.000000000 -0700
+++ ./tmp/Closure57fix/src/com/google/javascript/jscomp/ClosureCodingConvention.java	2019-05-05 07:50:56.000000000 -0700
@@ -194,7 +194,7 @@
         String qualifiedName = callee.getQualifiedName();
         if (functionName.equals(qualifiedName)) {
           Node target = callee.getNext();
-          if (target != null) {
+          if (target != null && target.getType() == Token.STRING) {
             className = target.getString();
           }
         }
