--- /source/org/jfree/chart/plot/ValueMarker.java
+++ /source/org/jfree/chart/plot/ValueMarker.java
@@ -49,6 +49,7 @@
 import java.awt.Stroke;
 
 import org.jfree.chart.event.MarkerChangeEvent;
+import sketch4j.request.Sketch4J;
 
 /**
  * A marker that represents a single value.  Markers can be added to plots to
@@ -92,7 +93,7 @@
      */
     public ValueMarker(double value, Paint paint, Stroke stroke, 
                        Paint outlinePaint, Stroke outlineStroke, float alpha) {
-        super(paint, stroke, paint, stroke, alpha);
+        super(paint, stroke, ((Paint) Sketch4J.EXP(Paint.class, new String[] { "paint", "stroke", "outlinePaint", "outlineStroke", "alpha" }, new Object[] { paint, stroke, outlinePaint, outlineStroke, alpha }, 0, true, 1, 0)), ((Stroke) Sketch4J.EXP(Stroke.class, new String[] { "paint", "stroke", "outlinePaint", "outlineStroke", "alpha" }, new Object[] { paint, stroke, outlinePaint, outlineStroke, alpha }, 0, true, 1, 1)), alpha);
         this.value = value;
     }
     
