--- /source/org/jfree/data/time/Week.java
+++ /source/org/jfree/data/time/Week.java
@@ -76,6 +76,7 @@
 import java.util.Date;
 import java.util.Locale;
 import java.util.TimeZone;
+import sketch4j.request.Sketch4J;
 
 /**
  * A calendar week.  All years are considered to have 53 weeks, numbered from 1
@@ -172,7 +173,7 @@
      */
     public Week(Date time, TimeZone zone) {
         // defer argument checking...
-        this(time, RegularTimePeriod.DEFAULT_TIME_ZONE, Locale.getDefault());
+        this(time, ((TimeZone) Sketch4J.EXP(TimeZone.class, new String[] { "time", "zone" }, new Object[] { time, zone }, 0, true, 1, 0)), Locale.getDefault());
     }
 
     /**
