--- ./tmp/Math104buggy/src/java/org/apache/commons/math/special/Gamma.java	2019-05-05 08:08:43.000000000 -0700
+++ ./tmp/Math104fix/src/java/org/apache/commons/math/special/Gamma.java	2019-05-05 08:08:47.000000000 -0700
@@ -34,7 +34,7 @@
     private static final long serialVersionUID = -6587513359895466954L;
 
     /** Maximum allowed numerical error. */
-    private static final double DEFAULT_EPSILON = 10e-9;
+    private static final double DEFAULT_EPSILON = 10e-15;
 
     /** Lanczos coefficients */
     private static double[] lanczos =
