--- /source/org/jfree/data/time/TimeSeries.java
+++ /source/org/jfree/data/time/TimeSeries.java
@@ -576,6 +576,7 @@
         if (item == null) {
             throw new IllegalArgumentException("Null 'item' argument.");
         }
+        findBoundsByIteration();
         item = (TimeSeriesDataItem) item.clone();
         Class c = item.getPeriod().getClass();
         if (this.timePeriodClass == null) {