--- ./tmp/Math6buggy/src/main/java/org/apache/commons/math3/optim/BaseOptimizer.java	2019-05-05 08:00:34.000000000 -0700
+++ ./tmp/Math6fix/src/main/java/org/apache/commons/math3/optim/BaseOptimizer.java	2019-05-05 08:00:39.000000000 -0700
@@ -48,7 +48,7 @@
         this.checker = checker;
 
         evaluations = new Incrementor(0, new MaxEvalCallback());
-        iterations = new Incrementor(0, new MaxIterCallback());
+        iterations = new Incrementor(Integer.MAX_VALUE, new MaxIterCallback());
     }
 
     /**
