--- /src/main/java/org/apache/commons/math/optimization/linear/SimplexSolver.java
+++ /src/main/java/org/apache/commons/math/optimization/linear/SimplexSolver.java
@@ -20,6 +20,7 @@
 import org.apache.commons.math.optimization.OptimizationException;
 import org.apache.commons.math.optimization.RealPointValuePair;
 import org.apache.commons.math.util.MathUtils;
+import sketch4j.request.Sketch4J;
 
 
 /**
@@ -79,7 +80,7 @@
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getHeight(); i++) {
             final double rhs = tableau.getEntry(i, tableau.getWidth() - 1);
             final double entry = tableau.getEntry(i, col);
-            if (MathUtils.compareTo(entry, 0, epsilon) >= 0) {
+            if (Sketch4J.EXPOP(int.class, new String[] { "MathUtils.compareTo(entry, 0, epsilon)", "0" }, new Object[] { MathUtils.compareTo(entry, 0, epsilon), 0 }, 0, true, 1, 0)) {
                 final double ratio = rhs / entry;
                 if (ratio < minRatio) {
                     minRatio = ratio;
