--- /src/main/java/org/apache/commons/math/analysis/solvers/BisectionSolver.java	
+++ /src/main/java/org/apache/commons/math/analysis/solvers/BisectionSolver.java
@@ -19,6 +19,7 @@
 import org.apache.commons.math.FunctionEvaluationException;
 import org.apache.commons.math.MaxIterationsExceededException;
 import org.apache.commons.math.analysis.UnivariateRealFunction;
+import sketch4j.request.Sketch4J;
 
 /**
  * Implements the <a href="http://mathworld.wolfram.com/Bisection.html">
@@ -69,7 +70,8 @@
     /** {@inheritDoc} */
     public double solve(final UnivariateRealFunction f, double min, double max, double initial)
         throws MaxIterationsExceededException, FunctionEvaluationException {
-        return solve(min, max);
+        return solve(((UnivariateRealFunction) Sketch4J.EXP(UnivariateRealFunction.class, new String[] { "f" }, new Object[] { f }, 0, true, 1, 0)), min, max);
+
     }
 
     /** {@inheritDoc} */
