--- ./tmp/Closure31buggy/src/com/google/javascript/jscomp/Compiler.java	2019-05-05 07:50:06.000000000 -0700
+++ ./tmp/Closure31fix/src/com/google/javascript/jscomp/Compiler.java	2019-05-05 07:50:10.000000000 -0700
@@ -1282,7 +1282,6 @@
 
       // Check if the sources need to be re-ordered.
       if (options.dependencyOptions.needsManagement() &&
-          !options.skipAllPasses &&
           options.closurePass) {
         for (CompilerInput input : inputs) {
           // Forward-declare all the provided types, so that they
