--- /source///org/jfree/data/time/TimeSeries.java
+++ /source///org/jfree/data/time/TimeSeries.java
@@ -624,7 +624,7 @@
             }
         }
         if (added) {
-            updateBoundsForAddedItem(item);
+            findBoundsByIteration();
             // check if this addition will exceed the maximum item count...
             if (getItemCount() > this.maximumItemCount) {
                 TimeSeriesDataItem d = (TimeSeriesDataItem) this.data.remove(0);
