--- /source///org/jfree/chart/plot/CategoryPlot.java
+++ /source///org/jfree/chart/plot/CategoryPlot.java
@@ -2640,9 +2640,6 @@
                 }
             }
 
-            for (int i = 0; i < datasetCount; i++) {
-                foundData = render(g2, dataArea, i, state) || foundData;
-            }
 
             // draw foreground annotations
             for (int i = 0; i < datasetCount; i++) {
