--- /source///org/jfree/chart/axis/AxisCollection.java
+++ /source///org/jfree/chart/axis/AxisCollection.java
@@ -129,18 +129,6 @@
         if (edge == null) {
             throw new IllegalArgumentException("Null 'edge' argument.");   
         }
-        if (edge == RectangleEdge.TOP) {
-            this.axesAtTop.add(axis);
-        }
-        else if (edge == RectangleEdge.BOTTOM) {
-            this.axesAtBottom.add(axis);
-        }
-        else if (edge == RectangleEdge.LEFT) {
-            this.axesAtLeft.add(axis);
-        }
-        else if (edge == RectangleEdge.RIGHT) {
-            this.axesAtRight.add(axis);
-        }
     }
 
 }
