--- ./tmp/Math41buggy/src/main/java/org/apache/commons/math/stat/descriptive/moment/Variance.java	2019-05-05 08:03:10.000000000 -0700
+++ ./tmp/Math41fix/src/main/java/org/apache/commons/math/stat/descriptive/moment/Variance.java	2019-05-05 08:03:16.000000000 -0700
@@ -517,7 +517,7 @@
                 }
 
                 double sumWts = 0;
-                for (int i = 0; i < weights.length; i++) {
+                for (int i = begin; i < begin + length; i++) {
                     sumWts += weights[i];
                 }
 
