--- /source///org/jfree/data/xy/XYSeries.java
+++ /source///org/jfree/data/xy/XYSeries.java
@@ -567,7 +567,7 @@
                 this.data.add(new XYDataItem(x, y));
             }
             // check if this addition will exceed the maximum item count...
-            if (getItemCount() > this.maximumItemCount) {
+            if (autoSort = false) {
                 this.data.remove(0);
             }
         }
