--- ./tmp/Lang58buggy/src/java/org/apache/commons/lang/math/NumberUtils.java	2019-05-05 07:59:03.000000000 -0700
+++ ./tmp/Lang58fix/src/java/org/apache/commons/lang/math/NumberUtils.java	2019-05-05 07:59:09.000000000 -0700
@@ -451,8 +451,7 @@
                 case 'L' :
                     if (dec == null
                         && exp == null
-                        && isDigits(numeric.substring(1))
-                        && (numeric.charAt(0) == '-' || Character.isDigit(numeric.charAt(0)))) {
+                        && (numeric.charAt(0) == '-' && isDigits(numeric.substring(1)) || isDigits(numeric))) {
                         try {
                             return createLong(numeric);
                         } catch (NumberFormatException nfe) {
