--- ./tmp/Math35buggy/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java	2019-05-05 08:02:36.000000000 -0700
+++ ./tmp/Math35fix/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java	2019-05-05 08:02:42.000000000 -0700
@@ -48,7 +48,7 @@
                                    final int populationLimit,
                                    final double elitismRate) {
         super(chromosomes, populationLimit);
-        this.elitismRate = elitismRate;
+        setElitismRate(elitismRate);
     }
 
     /**
@@ -62,7 +62,7 @@
      */
     public ElitisticListPopulation(final int populationLimit, final double elitismRate) {
         super(populationLimit);
-        this.elitismRate = elitismRate;
+        setElitismRate(elitismRate);
     }
 
     /**
