--- ./tmp/Mockito38buggy/src/org/mockito/internal/verification/argumentmatching/ArgumentMatchingTool.java	2019-05-05 08:10:00.000000000 -0700
+++ ./tmp/Mockito38fix/src/org/mockito/internal/verification/argumentmatching/ArgumentMatchingTool.java	2019-05-05 08:10:04.000000000 -0700
@@ -45,6 +45,6 @@
     }
 
     private boolean toStringEquals(Matcher m, Object arg) {
-        return StringDescription.toString(m).equals(arg.toString());
+        return StringDescription.toString(m).equals(arg == null? "null" : arg.toString());
     }
 }
