--- ./tmp/Math5buggy/src/main/java/org/apache/commons/math3/complex/Complex.java	2019-05-05 08:00:23.000000000 -0700
+++ ./tmp/Math5fix/src/main/java/org/apache/commons/math3/complex/Complex.java	2019-05-05 08:00:28.000000000 -0700
@@ -302,7 +302,7 @@
         }
 
         if (real == 0.0 && imaginary == 0.0) {
-            return NaN;
+            return INF;
         }
 
         if (isInfinite) {
