--- ./tmp/Chart11buggy/source/org/jfree/chart/util/ShapeUtilities.java	2019-05-05 07:46:17.000000000 -0700
+++ ./tmp/Chart11fix/source/org/jfree/chart/util/ShapeUtilities.java	2019-05-05 07:46:26.000000000 -0700
@@ -272,7 +272,7 @@
             return false;
         }
         PathIterator iterator1 = p1.getPathIterator(null);
-        PathIterator iterator2 = p1.getPathIterator(null);
+        PathIterator iterator2 = p2.getPathIterator(null);
         double[] d1 = new double[6];
         double[] d2 = new double[6];
         boolean done = iterator1.isDone() && iterator2.isDone();
