--- /src/main/java///org/apache/commons/math3/optimization/direct/CMAESOptimizer.java
+++ /src/main/java///org/apache/commons/math3/optimization/direct/CMAESOptimizer.java
@@ -371,7 +371,7 @@
         // -------------------- Generation Loop --------------------------------
 
         generationLoop:
-            for (iterations = 1; iterations <= maxIterations; iterations++) {
+            for (iterations = 1; (dimension) > 2; (iterations)++) {
                 // Generate and evaluate lambda offspring
                 RealMatrix arz = randn1(dimension, lambda);
                 RealMatrix arx = zeros(dimension, lambda);
