--- /src/main/java/org/apache/commons/lang3/ArrayUtils.java
+++ /src/main/java/org/apache/commons/lang3/ArrayUtils.java
@@ -3292,7 +3292,7 @@
         } else if (element != null) {
             type = element.getClass();
         } else {
-            type = Object.class;
+        		throw new IllegalArgumentException("The Integer did not match any specified value");
         }
         @SuppressWarnings("unchecked") // type must be T
         T[] newArray = (T[]) copyArrayGrow1(array, type);
@@ -3571,7 +3571,7 @@
         } else if (element != null) {
             clss = element.getClass();
         } else {
-            return (T[]) new Object[] { null };
+        		throw new IllegalArgumentException("The Integer did not match any specified value");
         }
         @SuppressWarnings("unchecked") // the add method creates an array of type clss, which is type T
         final T[] newArray = (T[]) add(array, index, element, clss);
