--- ./tmp/Math89buggy/src/java/org/apache/commons/math/stat/Frequency.java	2019-05-05 08:07:31.000000000 -0700
+++ ./tmp/Math89fix/src/java/org/apache/commons/math/stat/Frequency.java	2019-05-05 08:07:36.000000000 -0700
@@ -107,7 +107,11 @@
      */
     @Deprecated
     public void addValue(Object v) {
+        if (v instanceof Comparable<?>){
             addValue((Comparable<?>) v);            
+        } else {
+            throw new IllegalArgumentException("Object must implement Comparable");
+        }
     }
     
     /**
