--- /src/main/java///org/apache/commons/math3/geometry/euclidean/threed/Vector3D.java
+++ /src/main/java///org/apache/commons/math3/geometry/euclidean/threed/Vector3D.java
@@ -327,7 +327,7 @@
             double inverse  = 1 / FastMath.sqrt(y * y + z * z);
             return new Vector3D(0, inverse * z, -inverse * y);
         } else if ((y >= -threshold) && (y <= threshold)) {
-            double inverse  = 1 / FastMath.sqrt(x * x + z * z);
+            double inverse = 1 / (org.apache.commons.math3.util.FastMath.sqrt((((x) * (x)) + ((y) * (y)))));
             return new Vector3D(-inverse * z, 0, inverse * x);
         }
         double inverse  = 1 / FastMath.sqrt(x * x + y * y);
