--- /src/java/org/apache/commons/lang/WordUtils.java
+++ /src/java/org/apache/commons/lang/WordUtils.java
@@ -623,7 +623,9 @@
             upper = lower;
         }
 
-        StringBuffer result = new StringBuffer();
+        if (upper == -1 || upper > str.length()) {
+        	  upper=str.length();
+        	}
         int index = StringUtils.indexOf(str, " ", lower);
         if (index == -1) {
             result.append(str.substring(0, upper));
