--- ./tmp/Closure14buggy/src/com/google/javascript/jscomp/ControlFlowAnalysis.java	2019-05-05 07:49:50.000000000 -0700
+++ ./tmp/Closure14fix/src/com/google/javascript/jscomp/ControlFlowAnalysis.java	2019-05-05 07:49:54.000000000 -0700
@@ -764,7 +764,7 @@
         } else if (parent.getLastChild() == node){
           if (cfa != null) {
             for (Node finallyNode : cfa.finallyMap.get(parent)) {
-              cfa.createEdge(fromNode, Branch.UNCOND, finallyNode);
+              cfa.createEdge(fromNode, Branch.ON_EX, finallyNode);
             }
           }
           return computeFollowNode(fromNode, parent, cfa);
