--- ./tmp/Mockito29buggy/src/org/mockito/internal/matchers/Same.java	2019-05-05 08:09:43.000000000 -0700
+++ ./tmp/Mockito29fix/src/org/mockito/internal/matchers/Same.java	2019-05-05 08:09:47.000000000 -0700
@@ -26,7 +26,7 @@
     public void describeTo(Description description) {
         description.appendText("same(");
         appendQuoting(description);
-        description.appendText(wanted.toString());
+        description.appendText(wanted == null ? "null" : wanted.toString());
         appendQuoting(description);
         description.appendText(")");
     }
