--- ./tmp/Chart3buggy/source/org/jfree/data/time/TimeSeries.java	2019-05-05 07:44:25.000000000 -0700
+++ ./tmp/Chart3fix/source/org/jfree/data/time/TimeSeries.java	2019-05-05 07:44:35.000000000 -0700
@@ -1054,6 +1054,8 @@
             throw new IllegalArgumentException("Requires start <= end.");
         }
         TimeSeries copy = (TimeSeries) super.clone();
+        copy.minY = Double.NaN;
+        copy.maxY = Double.NaN;
         copy.data = new java.util.ArrayList();
         if (this.data.size() > 0) {
             for (int index = start; index <= end; index++) {
