--- ./tmp/Mockito8buggy/src/org/mockito/internal/util/reflection/GenericMetadataSupport.java	2019-05-05 08:09:12.000000000 -0700
+++ ./tmp/Mockito8fix/src/org/mockito/internal/util/reflection/GenericMetadataSupport.java	2019-05-05 08:09:17.000000000 -0700
@@ -76,7 +76,7 @@
 
             if (actualTypeArgument instanceof WildcardType) {
                 contextualActualTypeParameters.put(typeParameter, boundsOf((WildcardType) actualTypeArgument));
-            } else {
+            } else if (typeParameter != actualTypeArgument) {
                 contextualActualTypeParameters.put(typeParameter, actualTypeArgument);
             }
             // logger.log("For '" + parameterizedType + "' found type variable : { '" + typeParameter + "(" + System.identityHashCode(typeParameter) + ")" + "' : '" + actualTypeArgument + "(" + System.identityHashCode(typeParameter) + ")" + "' }");
