--- /source/org/jfree/chart/plot/CategoryPlot.java
+++ /source/org/jfree/chart/plot/CategoryPlot.java
@@ -2163,6 +2163,9 @@
             markers = (ArrayList) this.backgroundDomainMarkers.get(new Integer(
                     index));
         }
+//ACS's patch begin
+	if (markers == null){return false;}
+//ACS's patch end
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
@@ -2445,6 +2448,9 @@
             markers = (ArrayList) this.backgroundRangeMarkers.get(new Integer(
                     index));
         }
+//ACS's patch begin
+	if (markers == null){return false;}
+//ACS's patch end
         boolean removed = markers.remove(marker);
         if (removed && notify) {
             fireChangeEvent();
