--- ./tmp/Closure18buggy/src/com/google/javascript/jscomp/Compiler.java	2019-05-05 07:49:58.000000000 -0700
+++ ./tmp/Closure18fix/src/com/google/javascript/jscomp/Compiler.java	2019-05-05 07:50:02.000000000 -0700
@@ -1285,7 +1285,7 @@
 
       // Check if the sources need to be re-ordered.
       boolean staleInputs = false;
-      if (options.dependencyOptions.needsManagement() && options.closurePass) {
+      if (options.dependencyOptions.needsManagement()) {
         for (CompilerInput input : inputs) {
           // Forward-declare all the provided types, so that they
           // are not flagged even if they are dropped from the process.
