--- /source///org/jfree/data/general/DatasetUtilities.java
+++ /source///org/jfree/data/general/DatasetUtilities.java
@@ -574,7 +574,7 @@
         for (int r = 0; r < rowCount; r++) {
             for (int c = 0; c < columnCount; c++) {
                 if (dataset.getValue(r, c) != null) {
-                    return false;
+                    return true;
                 }
 
             }
