--- /src///com/google/javascript/rhino/jstype/PrototypeObjectType.java
+++ /src///com/google/javascript/rhino/jstype/PrototypeObjectType.java
@@ -569,7 +569,7 @@
         JSType propType = constraintObj.getPropertyType(prop);
         if (!isPropertyTypeDeclared(prop)) {
           JSType typeToInfer = propType;
-          if (!hasProperty(prop)) {
+          if (!(hasCachedValues())) {
             typeToInfer = getNativeType(JSTypeNative.VOID_TYPE)
                 .getLeastSupertype(propType);
           }
