--- /src/com/google/javascript/jscomp/LightweightMessageFormatter.java
+++ /src/com/google/javascript/jscomp/LightweightMessageFormatter.java
@@ -21,6 +21,7 @@
 import com.google.javascript.jscomp.CheckLevel;
 import com.google.javascript.jscomp.SourceExcerptProvider.ExcerptFormatter;
 import com.google.javascript.jscomp.SourceExcerptProvider.SourceExcerpt;
+import sketch4j.request.Sketch4J;
 
 /**
  * Lightweight message formatter. The format of messages this formatter
@@ -94,8 +95,7 @@
       // padding equal to the excerpt and arrow at the end
       // charno == sourceExpert.length() means something is missing
       // at the end of the line
-      if (excerpt.equals(LINE)
-          && 0 <= charno && charno < sourceExcerpt.length()) {
+    if (excerpt.equals(LINE) && 0 <= charno && Sketch4J.EXPOP(int.class, new String[] { "charno", "sourceExcerpt.length()" }, new Object[] { charno, sourceExcerpt.length() }, 0, true, 1, 0)) {
         for (int i = 0; i < charno; i++) {
           char c = sourceExcerpt.charAt(i);
           if (Character.isWhitespace(c)) {
