--- ./tmp/Lang33buggy/src/main/java/org/apache/commons/lang3/ClassUtils.java	2019-05-05 07:56:08.000000000 -0700
+++ ./tmp/Lang33fix/src/main/java/org/apache/commons/lang3/ClassUtils.java	2019-05-05 07:56:15.000000000 -0700
@@ -907,7 +907,7 @@
         }
         Class<?>[] classes = new Class[array.length];
         for (int i = 0; i < array.length; i++) {
-            classes[i] = array[i].getClass();
+            classes[i] = array[i] == null ? null : array[i].getClass();
         }
         return classes;
     }
