--- ./tmp/Math20buggy/src/main/java/org/apache/commons/math3/optimization/direct/CMAESOptimizer.java	2019-05-05 08:01:08.000000000 -0700
+++ ./tmp/Math20fix/src/main/java/org/apache/commons/math3/optimization/direct/CMAESOptimizer.java	2019-05-05 08:01:13.000000000 -0700
@@ -918,7 +918,8 @@
          * @return the original objective variables, possibly repaired.
          */
         public double[] repairAndDecode(final double[] x) {
-            return
+            return boundaries != null && isRepairMode ?
+                decode(repair(x)) :
                 decode(x);
         }
 
