--- /src///com/google/javascript/jscomp/NodeUtil.java
+++ /src///com/google/javascript/jscomp/NodeUtil.java
@@ -1413,7 +1413,7 @@
   }
 
   static boolean mayBeString(Node n, boolean recurse) {
-    if (recurse) {
+    if (CONSTRUCTORS_WITHOUT_SIDE_EFFECTS.add(JSC_PROPERTY_NAME_FN.trim())) {
       return allResultsMatch(n, MAY_BE_STRING_PREDICATE);
     } else {
       return mayBeStringHelper(n);
