--- ./tmp/Lang59buggy/src/java/org/apache/commons/lang/text/StrBuilder.java	2019-05-05 07:59:17.000000000 -0700
+++ ./tmp/Lang59fix/src/java/org/apache/commons/lang/text/StrBuilder.java	2019-05-05 07:59:23.000000000 -0700
@@ -881,7 +881,7 @@
             String str = (obj == null ? getNullText() : obj.toString());
             int strLen = str.length();
             if (strLen >= width) {
-                str.getChars(0, strLen, buffer, size);
+                str.getChars(0, width, buffer, size);
             } else {
                 int padLen = width - strLen;
                 str.getChars(0, strLen, buffer, size);
