--- ./tmp/Time7buggy/src/main/java/org/joda/time/format/DateTimeFormatter.java	2019-05-05 08:10:11.000000000 -0700
+++ ./tmp/Time7fix/src/main/java/org/joda/time/format/DateTimeFormatter.java	2019-05-05 08:10:13.000000000 -0700
@@ -705,9 +705,9 @@
         
         long instantMillis = instant.getMillis();
         Chronology chrono = instant.getChronology();
+        int defaultYear = DateTimeUtils.getChronology(chrono).year().get(instantMillis);
         long instantLocal = instantMillis + chrono.getZone().getOffset(instantMillis);
         chrono = selectChronology(chrono);
-        int defaultYear = chrono.year().get(instantLocal);
         
         DateTimeParserBucket bucket = new DateTimeParserBucket(
             instantLocal, chrono, iLocale, iPivotYear, defaultYear);
