--- ./tmp/Closure73buggy/src/com/google/javascript/jscomp/CodeGenerator.java	2019-05-05 07:51:43.000000000 -0700
+++ ./tmp/Closure73fix/src/com/google/javascript/jscomp/CodeGenerator.java	2019-05-05 07:51:47.000000000 -0700
@@ -1042,7 +1042,7 @@
             // No charsetEncoder provided - pass straight latin characters
             // through, and escape the rest.  Doing the explicit character
             // check is measurably faster than using the CharsetEncoder.
-            if (c > 0x1f && c <= 0x7f) {
+            if (c > 0x1f && c < 0x7f) {
               sb.append(c);
             } else {
               // Other characters can be misinterpreted by some js parsers,
