--- ./tmp/Mockito24buggy/src/org/mockito/internal/stubbing/defaultanswers/ReturnsEmptyValues.java	2019-05-05 08:09:23.000000000 -0700
+++ ./tmp/Mockito24fix/src/org/mockito/internal/stubbing/defaultanswers/ReturnsEmptyValues.java	2019-05-05 08:09:28.000000000 -0700
@@ -73,7 +73,7 @@
             //see issue 184.
             //mocks by default should return 0 if references are the same, otherwise some other value because they are not the same. Hence we return 1 (anything but 0 is good).
             //Only for compareTo() method by the Comparable interface
-            return 1;
+            return invocation.getMock() == invocation.getArguments()[0] ? 0 : 1;
         }
         
         Class<?> returnType = invocation.getMethod().getReturnType();
