--- /source/org/jfree/chart/renderer/GrayPaintScale.java
+++ /source/org/jfree/chart/renderer/GrayPaintScale.java
@@ -50,6 +50,7 @@
 import java.io.Serializable;
 
 import org.jfree.chart.util.PublicCloneable;
+import sketch4j.request.Sketch4J;
 
 /**
  * A paint scale that returns shades of gray.
@@ -123,8 +124,7 @@
     public Paint getPaint(double value) {
         double v = Math.max(value, this.lowerBound);
         v = Math.min(v, this.upperBound);
-        int g = (int) ((value - this.lowerBound) / (this.upperBound 
-                - this.lowerBound) * 255.0);
+        int g = (int) ((((Double) Sketch4J.EXP(double.class, new String[] { "v", "value", "upperBound", "lowerBound" }, new Object[] { v, value, upperBound, lowerBound }, 0, true, 1, 0)) - this.lowerBound) / (this.upperBound - this.lowerBound) * 255.0);
         return new Color(g, g, g);
     }
     
