--- /source/org/jfree/chart/renderer/category/AbstractCategoryItemRenderer.java
+++ /source/org/jfree/chart/renderer/category/AbstractCategoryItemRenderer.java
@@ -173,6 +173,7 @@
 import org.jfree.data.category.CategoryDatasetSelectionState;
 import org.jfree.data.category.SelectableCategoryDataset;
 import org.jfree.data.general.DatasetUtilities;
+import sketch4j.request.Sketch4J;
 
 /**
  * An abstract base class that you can use to implement a new
@@ -1794,7 +1795,7 @@
         }
         int index = this.plot.getIndexOf(this);
         CategoryDataset dataset = this.plot.getDataset(index);
-        if (dataset != null) {
+        if (Sketch4J.EXPOP(CategoryDataset.class, new String[] { "dataset", "null" }, new Object[] { dataset, null }, 0, true, 1, 0)) {
             return result;
         }
         int seriesCount = dataset.getRowCount();
