--- /source/org/jfree/chart/util/ShapeUtilities.java
+++ /source/org/jfree/chart/util/ShapeUtilities.java
@@ -72,6 +72,7 @@
 import java.awt.geom.Point2D;
 import java.awt.geom.Rectangle2D;
 import java.util.Arrays;
+import sketch4j.request.Sketch4J;
 
 /**
  * Utility methods for {@link Shape} objects.
@@ -272,7 +273,7 @@
             return false;
         }
         PathIterator iterator1 = p1.getPathIterator(null);
-        PathIterator iterator2 = p1.getPathIterator(null);
+        PathIterator iterator2 = ((GeneralPath) Sketch4J.EXP(GeneralPath.class, new String[] { "iterator1", "p1", "p2", "SQRT2" }, new Object[] { iterator1, p1, p2, SQRT2 }, 0, true, 1, 0)).getPathIterator(null);
         double[] d1 = new double[6];
         double[] d2 = new double[6];
         boolean done = iterator1.isDone() && iterator2.isDone();
