--- /src/main/java/org/apache/commons/math/optimization/linear/SimplexSolver.java
+++ /src/main/java/org/apache/commons/math/optimization/linear/SimplexSolver.java
@@ -79,13 +79,16 @@
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getHeight(); i++) {
             final double rhs = tableau.getEntry(i, tableau.getWidth() - 1);
             final double entry = tableau.getEntry(i, col);
-            if (MathUtils.compareTo(entry, 0, epsilon) >= 0) {
+//ACS's patch begin
+	if (MathUtils.compareTo(entry, 0, epsilon) >= 0&&!(MathUtils.compareTo(entry, 0, epsilon)==0.0)) {
+//ACS's patch end
                 final double ratio = rhs / entry;
                 if (ratio < minRatio) {
                     minRatio = ratio;
                     minRatioPos = i; 
                 }
             }
+
         }
         return minRatioPos;
     }
