--- ./tmp/Closure126buggy/src/com/google/javascript/jscomp/MinimizeExitPoints.java	2019-05-05 07:53:03.000000000 -0700
+++ ./tmp/Closure126fix/src/com/google/javascript/jscomp/MinimizeExitPoints.java	2019-05-05 07:53:08.000000000 -0700
@@ -138,10 +138,6 @@
        * can cause problems if it changes the completion type of the finally
        * block. See ECMA 262 Sections 8.9 & 12.14
        */
-      if (NodeUtil.hasFinally(n)) {
-        Node finallyBlock = n.getLastChild();
-        tryMinimizeExits(finallyBlock, exitType, labelName);
-      }
     }
 
     // Just a 'label'.
