--- ./tmp/Math96buggy/src/java/org/apache/commons/math/complex/Complex.java	2019-05-05 08:08:14.000000000 -0700
+++ ./tmp/Math96fix/src/java/org/apache/commons/math/complex/Complex.java	2019-05-05 08:08:20.000000000 -0700
@@ -255,7 +255,7 @@
                 if (rhs.isNaN()) {
                     ret = this.isNaN();
                 } else {
-                    ret = (Double.doubleToRawLongBits(real) == Double.doubleToRawLongBits(rhs.getReal())) && (Double.doubleToRawLongBits(imaginary) == Double.doubleToRawLongBits(rhs.getImaginary())); 
+                    ret = (real == rhs.real) && (imaginary == rhs.imaginary); 
                 }
             } catch (ClassCastException ex) {
                 // ignore exception
