--- ./tmp/Math11buggy/src/main/java/org/apache/commons/math3/distribution/MultivariateNormalDistribution.java	2019-05-05 08:00:57.000000000 -0700
+++ ./tmp/Math11fix/src/main/java/org/apache/commons/math3/distribution/MultivariateNormalDistribution.java	2019-05-05 08:01:03.000000000 -0700
@@ -180,7 +180,7 @@
             throw new DimensionMismatchException(vals.length, dim);
         }
 
-        return FastMath.pow(2 * FastMath.PI, -dim / 2) *
+        return FastMath.pow(2 * FastMath.PI, -0.5 * dim) *
             FastMath.pow(covarianceMatrixDeterminant, -0.5) *
             getExponentTerm(vals);
     }
