--- ./tmp/Math33buggy/src/main/java/org/apache/commons/math3/optimization/linear/SimplexTableau.java	2019-05-05 08:02:14.000000000 -0700
+++ ./tmp/Math33fix/src/main/java/org/apache/commons/math3/optimization/linear/SimplexTableau.java	2019-05-05 08:02:20.000000000 -0700
@@ -335,7 +335,7 @@
         // positive cost non-artificial variables
         for (int i = getNumObjectiveFunctions(); i < getArtificialVariableOffset(); i++) {
             final double entry = tableau.getEntry(0, i);
-            if (Precision.compareTo(entry, 0d, maxUlps) > 0) {
+            if (Precision.compareTo(entry, 0d, epsilon) > 0) {
                 columnsToDrop.add(i);
             }
         }
