--- ./tmp/Math32buggy/src/main/java/org/apache/commons/math3/geometry/euclidean/twod/PolygonsSet.java	2019-05-05 08:02:03.000000000 -0700
+++ ./tmp/Math32fix/src/main/java/org/apache/commons/math3/geometry/euclidean/twod/PolygonsSet.java	2019-05-05 08:02:08.000000000 -0700
@@ -133,7 +133,7 @@
 
         if (v.length == 0) {
             final BSPTree<Euclidean2D> tree = getTree(false);
-            if ((Boolean) tree.getAttribute()) {
+            if (tree.getCut() == null && (Boolean) tree.getAttribute()) {
                 // the instance covers the whole space
                 setSize(Double.POSITIVE_INFINITY);
                 setBarycenter(Vector2D.NaN);
