--- ./tmp/Lang24buggy/src/main/java/org/apache/commons/lang3/math/NumberUtils.java	2019-05-05 07:55:10.000000000 -0700
+++ ./tmp/Lang24fix/src/main/java/org/apache/commons/lang3/math/NumberUtils.java	2019-05-05 07:55:15.000000000 -0700
@@ -1410,7 +1410,7 @@
             if (chars[i] == 'l'
                 || chars[i] == 'L') {
                 // not allowing L with an exponent or decimal point
-                return foundDigit && !hasExp;
+                return foundDigit && !hasExp && !hasDecPoint;
             }
             // last character is illegal
             return false;
