--- ./tmp/Math27buggy/src/main/java/org/apache/commons/math3/fraction/Fraction.java	2019-05-05 08:01:19.000000000 -0700
+++ ./tmp/Math27fix/src/main/java/org/apache/commons/math3/fraction/Fraction.java	2019-05-05 08:01:25.000000000 -0700
@@ -594,7 +594,7 @@
      * @return the fraction percentage as a <tt>double</tt>.
      */
     public double percentageValue() {
-        return multiply(100).doubleValue();
+        return 100 * doubleValue();
     }
 
     /**
