--- ./tmp/Chart26buggy/source/org/jfree/chart/axis/Axis.java	2019-05-05 07:49:24.000000000 -0700
+++ ./tmp/Chart26fix/source/org/jfree/chart/axis/Axis.java	2019-05-05 07:49:35.000000000 -0700
@@ -1189,11 +1189,13 @@
         }
         if (plotState != null && hotspot != null) {
             ChartRenderingInfo owner = plotState.getOwner();
+            if (owner != null) {
                 EntityCollection entities = owner.getEntityCollection();
                 if (entities != null) {
                     entities.add(new AxisLabelEntity(this, hotspot, 
                             this.labelToolTip, this.labelURL));
                 }
+            }
         }
         return state;
 
