--- ./buggy_projects/Math/math_41_buggy/src/main/java/org/apache/commons/math/stat/descriptive/moment/Variance.java	2018-08-14 02:07:56.000000000 -0700
+++ A.java	2018-08-14 02:13:08.000000000 -0700
@@ -517,7 +517,7 @@
                 }
 
                 double sumWts = 0;
-                for (int i = 0; i < weights.length; i++) {
+                for(int i=begin;i<begin+length;i++){
                     sumWts += weights[i];
                 }
 
