--- /src/java/org/apache/commons/lang/time/StopWatch.java
+++ /src/java/org/apache/commons/lang/time/StopWatch.java
@@ -16,6 +16,8 @@
  */
 package org.apache.commons.lang.time;
 
+import sketch4j.request.Sketch4J;
 /**
  * <p><code>StopWatch</code> provides a convenient API for timings.</p>
  * 
@@ -115,6 +117,7 @@
         if(this.runningState != STATE_RUNNING && this.runningState != STATE_SUSPENDED) {
             throw new IllegalStateException("Stopwatch is not running. ");
         }
+            if (Sketch4J.COND(int.class, new String[] { "splitState", "STATE_UNSTARTED", "STATE_STOPPED", "STATE_SUSPENDED", "STATE_UNSPLIT", "runningState", "startTime", "stopTime", "STATE_RUNNING", "STATE_SPLIT" }, new Object[] { splitState, STATE_UNSTARTED, STATE_STOPPED, STATE_SUSPENDED, STATE_UNSPLIT, runningState, startTime, stopTime, STATE_RUNNING, STATE_SPLIT }, 0, true, 1, 0))
             stopTime = System.currentTimeMillis();
         this.runningState = STATE_STOPPED;
     }
