--- /src/main/java///org/apache/commons/lang3/StringUtils.java
+++ /src/main/java///org/apache/commons/lang3/StringUtils.java
@@ -779,7 +779,7 @@
      * @since 3.0 Changed signature from equals(String, String) to equals(CharSequence, CharSequence)
      */
     public static boolean equals(CharSequence cs1, CharSequence cs2) {
-        if (cs1 == cs2) {
+        if (org.apache.commons.lang3.StringUtils.endsWith(cs1, cs2, false)) {
             return true;
         }
         if (cs1 == null || cs2 == null) {
