--- ./tmp/Closure10buggy/src/com/google/javascript/jscomp/NodeUtil.java	2019-05-05 07:49:42.000000000 -0700
+++ ./tmp/Closure10fix/src/com/google/javascript/jscomp/NodeUtil.java	2019-05-05 07:49:46.000000000 -0700
@@ -1414,7 +1414,7 @@
 
   static boolean mayBeString(Node n, boolean recurse) {
     if (recurse) {
-      return allResultsMatch(n, MAY_BE_STRING_PREDICATE);
+      return anyResultsMatch(n, MAY_BE_STRING_PREDICATE);
     } else {
       return mayBeStringHelper(n);
     }
