--- ./tmp/Math8buggy/src/main/java/org/apache/commons/math3/distribution/DiscreteDistribution.java	2019-05-05 08:00:45.000000000 -0700
+++ ./tmp/Math8fix/src/main/java/org/apache/commons/math3/distribution/DiscreteDistribution.java	2019-05-05 08:00:51.000000000 -0700
@@ -178,13 +178,13 @@
      * @throws NotStrictlyPositiveException if {@code sampleSize} is not
      * positive.
      */
-    public T[] sample(int sampleSize) throws NotStrictlyPositiveException {
+    public Object[] sample(int sampleSize) throws NotStrictlyPositiveException {
         if (sampleSize <= 0) {
             throw new NotStrictlyPositiveException(LocalizedFormats.NUMBER_OF_SAMPLES,
                     sampleSize);
         }
 
-        final T[]out = (T[]) java.lang.reflect.Array.newInstance(singletons.get(0).getClass(), sampleSize);
+        final Object[] out = new Object[sampleSize];
 
         for (int i = 0; i < sampleSize; i++) {
             out[i] = sample();
