--- /source///org/jfree/data/general/DatasetUtilities.java
+++ /source///org/jfree/data/general/DatasetUtilities.java
@@ -571,14 +571,6 @@
             return true;
         }
 
-        for (int r = 0; r < rowCount; r++) {
-            for (int c = 0; c < columnCount; c++) {
-                if (dataset.getValue(r, c) != null) {
-                    return false;
-                }
-
-            }
-        }
 
         return true;
 
