--- ./tmp/Time19buggy/src/main/java/org/joda/time/DateTimeZone.java	2019-05-05 08:10:35.000000000 -0700
+++ ./tmp/Time19fix/src/main/java/org/joda/time/DateTimeZone.java	2019-05-05 08:10:37.000000000 -0700
@@ -897,7 +897,7 @@
                     return offsetLocal;
                 }
             }
-        } else if (offsetLocal > 0) {
+        } else if (offsetLocal >= 0) {
             long prev = previousTransition(instantAdjusted);
             if (prev < instantAdjusted) {
                 int offsetPrev = getOffset(prev);
