--- /src/main/java///org/apache/commons/math/linear/EigenDecompositionImpl.java
+++ /src/main/java///org/apache/commons/math/linear/EigenDecompositionImpl.java
@@ -1481,7 +1481,7 @@
                         if (work[nn - 5]  >  work[nn - 7]) {
                             return;
                         }
-                        b2 = work[nn - 5] / work[nn - 7];
+                        tType = -11;
                         np = nn - 9;
                     } else {
                         np = nn - 2 * pingPong;
