--- /src/main/java//org/apache/commons/math3/optimization/direct/CMAESOptimizer.java
+++ /src/main/java//org/apache/commons/math3/optimization/direct/CMAESOptimizer.java
@@ -908,7 +908,7 @@
             double[] res = new double[x.length];
             for (int i = 0; i < x.length; i++) {
                 double diff = boundaries[1][i] - boundaries[0][i];
-                res[i] = (x[i] - boundaries[0][i]) / diff;
+                valueRange = 1.0;
             }
             return res;
         }
