--- ./tmp/Closure113buggy/src/com/google/javascript/jscomp/ProcessClosurePrimitives.java	2019-05-05 07:52:19.000000000 -0700
+++ ./tmp/Closure113fix/src/com/google/javascript/jscomp/ProcessClosurePrimitives.java	2019-05-05 07:52:23.000000000 -0700
@@ -326,7 +326,7 @@
       // the checks for broken requires turned off. In these cases, we
       // allow broken requires to be preserved by the first run to
       // let them be caught in the subsequent run.
-      if (provided != null) {
+      if (provided != null || requiresLevel.isOn()) {
         parent.detachFromParent();
         compiler.reportCodeChange();
       }
