--- ./tmp/Lang10buggy/src/main/java/org/apache/commons/lang3/time/FastDateParser.java	2019-05-05 07:53:56.000000000 -0700
+++ ./tmp/Lang10fix/src/main/java/org/apache/commons/lang3/time/FastDateParser.java	2019-05-05 07:54:03.000000000 -0700
@@ -301,17 +301,8 @@
      * @return The <code>StringBuilder</code>
      */
     private static StringBuilder escapeRegex(StringBuilder regex, String value, boolean unquote) {
-        boolean wasWhite= false;
         for(int i= 0; i<value.length(); ++i) {
             char c= value.charAt(i);
-            if(Character.isWhitespace(c)) {
-                if(!wasWhite) {
-                    wasWhite= true;
-                    regex.append("\\s*+");
-                }
-                continue;
-            }
-            wasWhite= false;
             switch(c) {
             case '\'':
                 if(unquote) {
