--- ./tmp/Lang43buggy/src/java/org/apache/commons/lang/text/ExtendedMessageFormat.java	2019-05-05 07:57:20.000000000 -0700
+++ ./tmp/Lang43fix/src/java/org/apache/commons/lang/text/ExtendedMessageFormat.java	2019-05-05 07:57:27.000000000 -0700
@@ -419,6 +419,7 @@
         int start = pos.getIndex();
         char[] c = pattern.toCharArray();
         if (escapingOn && c[start] == QUOTE) {
+            next(pos);
             return appendTo == null ? null : appendTo.append(QUOTE);
         }
         int lastHold = start;
