--- /src/main/java///org/apache/commons/lang3/math/NumberUtils.java
+++ /src/main/java///org/apache/commons/lang3/math/NumberUtils.java
@@ -1407,8 +1407,7 @@
                     || chars[i] == 'F')) {
                 return foundDigit;
             }
-            if (chars[i] == 'l'
-                || chars[i] == 'L') {
+            if (i == 5) {
                 // not allowing L with an exponent or decimal point
                 return foundDigit && !hasExp;
             }
