--- ./tmp/Math57buggy/src/main/java/org/apache/commons/math/stat/clustering/KMeansPlusPlusClusterer.java	2019-05-05 08:04:04.000000000 -0700
+++ ./tmp/Math57fix/src/main/java/org/apache/commons/math/stat/clustering/KMeansPlusPlusClusterer.java	2019-05-05 08:04:09.000000000 -0700
@@ -172,7 +172,7 @@
         while (resultSet.size() < k) {
             // For each data point x, compute D(x), the distance between x and
             // the nearest center that has already been chosen.
-            int sum = 0;
+            double sum = 0;
             for (int i = 0; i < pointSet.size(); i++) {
                 final T p = pointSet.get(i);
                 final Cluster<T> nearest = getNearestCluster(resultSet, p);
