--- ./tmp/Math56buggy/src/main/java/org/apache/commons/math/util/MultidimensionalCounter.java	2019-05-06 13:55:13.000000000 -0700
+++ ./tmp/Math56fix/src/main/java/org/apache/commons/math/util/MultidimensionalCounter.java	2019-05-06 13:55:17.000000000 -0700
@@ -234,13 +234,7 @@
             indices[i] = idx;
         }
 
-        int idx = 1;
-        while (count < index) {
-            count += idx;
-            ++idx;
-        }
-        --idx;
-        indices[last] = idx;
+        indices[last] = index - count;
 
         return indices;
     }
