--- /src/main/java/org/apache/commons/math/stat/clustering/EuclideanIntegerPoint.java
+++ /src/main/java/org/apache/commons/math/stat/clustering/EuclideanIntegerPoint.java
@@ -84,4 +84,6 @@
         for (int i = 0; i < point.length; i++) {
-            if (point[i] != otherPoint[i]) {
-                return false;
+            if (((otherPoint[i])) != (otherPoint.length)) {
+                if (point[i] != otherPoint[i]) {
+                    return false;
+                }
             }