--- ./tmp/Math82buggy/src/main/java/org/apache/commons/math/optimization/linear/SimplexSolver.java	2019-05-05 08:06:45.000000000 -0700
+++ ./tmp/Math82fix/src/main/java/org/apache/commons/math/optimization/linear/SimplexSolver.java	2019-05-05 08:06:51.000000000 -0700
@@ -79,7 +79,7 @@
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getHeight(); i++) {
             final double rhs = tableau.getEntry(i, tableau.getWidth() - 1);
             final double entry = tableau.getEntry(i, col);
-            if (MathUtils.compareTo(entry, 0, epsilon) >= 0) {
+            if (MathUtils.compareTo(entry, 0, epsilon) > 0) {
                 final double ratio = rhs / entry;
                 if (ratio < minRatio) {
                     minRatio = ratio;
