--- ./tmp/Math94buggy/src/java/org/apache/commons/math/util/MathUtils.java	2019-05-05 08:08:03.000000000 -0700
+++ ./tmp/Math94fix/src/java/org/apache/commons/math/util/MathUtils.java	2019-05-05 08:08:08.000000000 -0700
@@ -409,7 +409,7 @@
      * @since 1.1
      */
     public static int gcd(int u, int v) {
-        if (u * v == 0) {
+        if ((u == 0) || (v == 0)) {
             return (Math.abs(u) + Math.abs(v));
         }
         // keep u and v negative, as negative integers range down to
