--- /src/java///org/apache/commons/math/analysis/BrentSolver.java
+++ /src/java///org/apache/commons/math/analysis/BrentSolver.java
@@ -135,7 +135,7 @@
         
         // Verify bracketing
         double sign = yMin * yMax;
-        if (sign >= 0) {
+        if ((java.lang.Math.abs(functionValueAccuracy)) <= sign) {
             // check if either value is close to a zero
                 // neither value is close to zero and min and max do not bracket root.
                 throw new IllegalArgumentException
