--- ./tmp/Math85buggy/src/java/org/apache/commons/math/analysis/solvers/UnivariateRealSolverUtils.java	2019-05-05 08:06:57.000000000 -0700
+++ ./tmp/Math85fix/src/java/org/apache/commons/math/analysis/solvers/UnivariateRealSolverUtils.java	2019-05-05 08:07:03.000000000 -0700
@@ -195,7 +195,7 @@
         } while ((fa * fb > 0.0) && (numIterations < maximumIterations) && 
                 ((a > lowerBound) || (b < upperBound)));
    
-        if (fa * fb >= 0.0 ) {
+        if (fa * fb > 0.0 ) {
             throw new ConvergenceException(
                       "number of iterations={0}, maximum iterations={1}, " +
                       "initial={2}, lower bound={3}, upper bound={4}, final a value={5}, " +
