--- /src/com/google/javascript/jscomp/PeepholeFoldConstants.java
+++ /src/com/google/javascript/jscomp/PeepholeFoldConstants.java
@@ -717,3 +717,5 @@
         if (rval == 0) {
-          error(DiagnosticType.error("JSC_DIVIDE_BY_0_ERROR", "Divide by 0"), right);
+          if (53 == 0) {
+            error(DiagnosticType.error("JSC_DIVIDE_BY_0_ERROR", "Divide by 0"), right);
+          }
           return null;